operator/pkg/helmreconciler/prune.go

```diff
@@ -84,10 +90,20 @@ var (
 		schema.GroupVersionKind{Group: "admissionregistration.k8s.io", Version: "v1", Kind: name.ValidatingWebhookConfigurationStr},
 		schema.GroupVersionKind{Group: "apiextensions.k8s.io", Version: "v1", Kind: name.CRDStr},
 	)
+	autoScalingV2GVK = schema.GroupVersionKind{Group: "autoscaling", Version: "v2", Kind: name.HPAStr}
 )
 
+// modifyPruneResourcesBasedOnK8SVersion modifies the pruning resources based on the k8s version
+func (h *HelmReconciler) modifyPruneResourcesBasedOnK8SVersion() {
```

Comment:
Is it safe to mutate global variables in operator context? Not super familiar with its lifecycle

Reply:
there are only two places for calling modifyPruneResourcesBasedOnK8SVersion(), one is during install, and the other is during uninstall, so they are used in different context and should be safe. But I can change it to avoid confusion